stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/usecase/SendMessageWithFilesTest.kt

```diff
@@ -145,40 +145,6 @@ internal class SendMessageWithFilesTest : BaseDomainTest2() {
         }
     }
 
-    @Test
-    fun `Upload attachment with error should be configurable`() = testCoroutines.scope.runBlockingTest {
-        val attachments = randomAttachmentsWithFile().toMutableList()
-        val files: List<File> = attachments.map { it.upload!! }
-
-        mockFileUploadsFailure(files)
-        val extra = mutableMapOf<String, Any>("The Answer" to 42)
-
-        for (attachment in attachments) {
-            val error = attachment.upload!!.toChatError()
-            val result = channelControllerImpl.uploadAttachment(attachment = attachment) { attachment, _ ->
-                attachment.copy(extraData = extra)
-            }
-            assertFailure(result)
-            Truth.assertThat(result.data().uploadState).isInstanceOf(Attachment.UploadState.Failed::class.java)
-            Truth.assertThat(result.data().extraData).isEqualTo(extra)
-        }
-    }
-
-    @Test
-    fun `Upload attachment should with errors should have the right format`() = testCoroutines.scope.runBlockingTest {
-        val attachments = randomAttachmentsWithFile().toMutableList()
-        val files: List<File> = attachments.map { it.upload!! }
-
-        mockFileUploadsFailure(files)
-
-        for (attachment in attachments) {
-            val error = attachment.upload!!.toChatError()
-            val result = channelControllerImpl.uploadAttachment(attachment = attachment)
-            assertFailure(result)
-            Truth.assertThat(result.data().uploadState).isInstanceOf(Attachment.UploadState.Failed::class.java)
```

Comment:
We should not be able to require data from a failed request, this is what caused this bug... I don't think that attachments with errors should have any format at all. The State is handled outside of the function `uploadAttachment`, in the `sendMessage` function.